### PR TITLE
Adding sorting of list in list view

### PR DESCRIPTION
**Pull request by jyotikasana**
Task details: https://github.com/amlcloud/screensite/issues/116

image

### CODE REVIEW

File: lib/lists/list_entitylistview.dart [hunk 1]
```diff
@@ -21,6 +23,13 @@ class EntityListView extends ConsumerStatefulWidget {
   ConsumerState<EntityListView> createState() => _EntityListViewState();
 }
 
+List<MapEntry<int, QueryDocumentSnapshot<Map<String, dynamic>>>> sortList(
+    List<MapEntry<int, QueryDocumentSnapshot<Map<String, dynamic>>>> list1) {
```
Comment (kovarny) [inline lib/lists/list_entitylistview.dart:27]: why naming parameter list1?

File: lib/lists/list_entitylistview.dart [hunk 2]
```diff
@@ -21,6 +23,13 @@ class EntityListView extends ConsumerStatefulWidget {
...
+  return list1
+    ..sort(
+        (e1, e2) => e1.value.data()['name'].compareTo(e2.value.data()['name']));
```
Comment (kovarny) [inline lib/lists/list_entitylistview.dart:30]: This should not rely on 'name' but instead rely on the entityName1 or data()['entityName1']+' '+data()['entityName2'] values...

image

**Review by kovarny — CHANGES_REQUESTED** [review]

File: lib/lists/list_entitylistview.dart (hunk 2 above)
Comment (jyotikasana) [reply lib/lists/list_entitylistview.dart:30]: Using entityname1 to sort as when using entityname1 and entityname2 , getting error in UI
```
 destinationList = sourceList
        ..sort((e1, e2) =>
            (e1.value.data()['entityName1'] + e1.value.data()['entityName2'])
                .compareTo(e2.value.data()['entityName1'] +
                    e2.value.data()['entityName2']));
```
image

Comment (jyotikasana) [reply lib/lists/list_entitylistview.dart:30]: Also, entityname1 and entityname2 fields are not present in the map we are sorting here, we either have name or entity number,
image

File: lib/lists/list_entitylistview.dart (hunk 1 above)
Comment (jyotikasana) [reply lib/lists/list_entitylistview.dart]: Updated parameter name to sourceList

**Review by chullman — CHANGES_REQUESTED** [review]
I think we want to do the sorting before the display of any of the list tiles in the 1st column, so my thoughts would be to do the sorting  on `lists_list.dart`

Right under Line 60 on that file you could add the following for case-insensitive search:

```dart
..sort((e1, e2) => e1[e1["entitiesName1"]]
.toLowerCase()
.compareTo(e2[e2["entitiesName1"]].toLowerCase())))
```
image

However, this will currently throw errors because in Firestore, if a field value for `entitiesName1`, e.g. `firstName` doesn't also have a `firstName` field with a value, there will be null field exceptions. Might need to ask Serge how to best solve this one.

**Comment by jyotikasana** [comment]
Added sorting in list by name field for below task.
https://github.com/amlcloud/screensite/issues/131

File: lib/lists/list_entitylistview.dart [hunk 3]
```diff
@@ -2,6 +2,7 @@ import 'package:cloud_firestore/cloud_firestore.dart';
 import 'package:flutter/material.dart';
 import 'package:flutter/services.dart';
 import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:http/http.dart';
 import 'package:screensite/providers/firestore.dart';
```
Comment (kovarny) [inline lib/lists/list_entitylistview.dart:6]: We should not make changes to this file for list sorting. It is another task

**Review by kovarny — CHANGES_REQUESTED** [review]

**Comment by jyotikasana** [comment]
Closing this PR to move on to other task, as discussed this task is now assigned to Chris.

**Pull request closed without merging**